### DISCUSSION TITLE
Cursor appearing over osu circle when increasing speed of movement

### DISCUSSION THREAD

**Discussion by Tarkan-Zarrouk** [ppy/osu#30336] [Q&A] [answered]
### Type

Cosmetic

### Bug description

Tablet: XPen G430s pro

After playing levels that are fast paced the cursor reveals itself and since it's a lot slower then the actual osu!lazer cursor it is very noticeable when playing, it's not a major issue but I do not believe that should be happening as it doesn't happen to my mouse (Some random office mouse I own)

Could this just be a tablet issue? It's my first tablet and I only recently discovered that this is not an ideal one but I do like it so I was just wondering if there's a possible fix to it or I should deal with it / switch to a newer tablet

### Screenshots or videos

https://github.com/user-attachments/assets/1904b4c2-ed0f-4b07-8411-55f4dae0a032

### Version

Lazer 2024.10091

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17425047/compressed-logs.zip)

**Comment by bdach** [comment]
If anything it's a macOS issue. The mouse cursor is not supposed to be showing in front of the game ever.

We can't fix this ourselves either, it's probably some issue in SDL, but reporting it there would require isolating this and I can't do that.

**Reply by Feodor0090** [reply]
there is a toggle in macos settings to enlarge the cursor when shaked, it serves the same purpose as windows' "ctrl to find cursor". Probably related here

**Reply by Joehuu** [reply]
Related: https://github.com/ppy/osu/issues/11772 and https://github.com/ppy/osu/issues/24620

**Reply by Tarkan-Zarrouk** [reply]
I had the shake to enlarge disabled for a while now but oh well that's alright I'll eventually find some possible solution that works with me thanks anyways 👍

**Reply by Tarkan-Zarrouk** [reply, marked as answer]
Fix for this: Switch from borderless to fullscreen or windowed

fullscreen: lose ability to alt-tab
windowed: smaller screen resolution

OR 

Borderless: Moving at high rate or moving near to the top of your screen will cause cursor to appear due to mac thinking you want to use your doc, the fix for this is to alt + tab out then alt-tab in (or simply reopen osu by clicking the icon)
